spec/alces_utils.rb

```diff
@@ -62,6 +47,35 @@ def nodeattr_cmd_trim_f(command)
       command.sub(AlcesUtils::GENDERS_FILE_REGEX, '')
     end
 
+    # Mocks nodeattr to use faked genders file
+    def spoof_nodeattr(context)
+      context.instance_exec do
+        genders_path = Metalware::FilePath.genders
+        genders_exist = File.exist? genders_path
+        File.write(genders_path, "local local\n") unless genders_exist
+
+        allow(Metalware::NodeattrInterface)
+          .to receive(:nodeattr).and_wrap_original do |method, *args|
+          AlcesUtils.check_and_raise_fakefs_error
+          path = AlcesUtils.nodeattr_genders_file_path(args[0])
+          cmd = AlcesUtils.nodeattr_cmd_trim_f(args[0])
+          genders_data = File.read(path)
+          tempfile = nil
+          begin
+            FakeFS.without do
+              tempfile = Tempfile.open('mock-genders')
+              tempfile.write(genders_data)
+              tempfile.close
```

Comment:
I believe you may be able to replace this block with `Tempfile.open('mock-genders') { |f| f.write(genders_data) }`, which is a bit simpler.

Reply:
In this situation we can not. The `tempfile` must be opened with `FakeFS` off. However in-order to ensure the file is indeed deleted, the file pointer `tempfile` needs to be unlinked after `nodeattr` is called (thus it is within the ensure block).

This means that the `tempfile` variable needs to be defined outside of the entire `being-ensure` block.

In lining it would result in the number of characters wrote to the file being returned NOT the file pointer thus breaking the unlink.

Reply:
tl;dr No because `FakeFS` is a pain in the ass

Reply:
Fair enough :slightly_smiling_face: